src/configuration.rs

```diff
@@ -1,80 +1,281 @@
+use std::str::FromStr;
```

Comment:
As we are targeting a library use-case at some point, it would make sense to start writing documentation in the module headers. For now, this is good though!

Reply:
Good point :ok_hand: I'll add this as an issue so it won't get forgotten about.